pkg/controller/directvolumemigration/rsync.go

```diff
@@ -565,6 +578,69 @@ func (t *Task) getRsyncRoute(namespace string) (string, error) {
 	return route.Spec.Host, nil
 }
 
+func (t *Task) createRsyncPassword() (string, error) {
+	var letters = []byte("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ")
+	random.Seed(time.Now().UnixNano())
+	password := make([]byte, 6)
+	for i := range password {
+		password[i] = letters[random.Intn(len(letters))]
+	}
+
+	secret := corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: migapi.OpenshiftMigrationNamespace,
+			Name:      "directvolumemigration-rsync-pass",
+			Labels: map[string]string{
+				"app": "directvolumemigration-rsync-transfer",
+			},
+		},
+		StringData: map[string]string{
+			corev1.BasicAuthPasswordKey: string(password),
+		},
+		Type: corev1.SecretTypeBasicAuth,
+	}
+	err := t.Client.Create(context.TODO(), &secret)
+	if k8serror.IsAlreadyExists(err) {
+		t.Log.Info("Secret already exists on host", "name", "directvolumemigration-rsync-pass", "namespace", migapi.OpenshiftMigrationNamespace)
+	} else if err != nil {
+		return "", err
+	}
+	return string(password), nil
+}
+
+func (t *Task) getRsyncPassword() (string, error) {
+	rsyncSecret := corev1.Secret{}
+	key := types.NamespacedName{Name: "directvolumemigration-rsync-pass", Namespace: migapi.OpenshiftMigrationNamespace}
```

Comment:
question: what would happen if two dvm CR are running at the same time, both of them will use the same password? Is this intentional or a mistake?

Reply:
This was not intentional. @dymurray @alaypatel07 is creating one secret per dvm for this better?

Reply:
I think eventually we will want to do that yes. Today we can reasonably expect only one DVM to be created and running at a time.

I also don't see a huge deal with reusing the same password since its all secured with certs anyway